plugins/backend/backstage-plugin-argo-cd-backend/README.md

```diff
@@ -16,16 +16,75 @@ argocd:
       instances:
         - name: argoInstance1
           url: https://argoInstance1.com
-          token: ${ARGOCD_AUTH_TOKEN}
+          token: ${ARGOCD_AUTH_TOKEN} # optional 
         - name: argoInstance2
           url: https://argoInstance2.com
+          # dedicated username/password for this instance
+          username: ${ARGOCD_USERNAME_INSTANCE_2} # optional 
+          password: ${ARGOCD_PASSWORD_INSTANCE_2} # optional 
 ```
 
-Add the required auth tokens to environmental variables, `ARGOCD_USERNAME` and `ARGOCD_PASSWORD`.
+### Authentication
+
+**Option 1**: Add the required auth tokens to environmental variables, `ARGOCD_USERNAME` and `ARGOCD_PASSWORD` inside the `argocd` object. It will be use as credentials for all instances by default.
+
+_Example_
+```yml
+argocd:
+  username: ${ARGOCD_USERNAME}
+  password: ${ARGOCD_PASSWORD}
+  appLocatorMethods:
+    - type: 'config'
+      instances:
+        - name: argoInstance1
+          url: https://argoInstance1.com
+        - name: argoInstance2
+          url: https://argoInstance2.com
+```
+
+**Option 2**: Define a `username` and a `password` for each instance. It has an higher priority than **Option 1**.
+
+_Example_
+```yml
+argocd:
+  username: ${ARGOCD_USERNAME} # Now only use for instance 1
+  password: ${ARGOCD_PASSWORD} # Now only use for instance 1
+  appLocatorMethods:
+    - type: 'config'
+      instances:
+        - name: argoInstance1
+          url: https://argoInstance1.com
+        - name: argoInstance2
+          url: https://argoInstance2.com
+          # dedicated username/password for this instance
+          username: ${ARGOCD_USERNAME_INSTANCE_2}
+          password: ${ARGOCD_PASSWORD_INSTANCE_2}
+```
+
+**Option 3**: Define a `token` for each instance. It has an higher priority than **Option 1** and **Option 2**.
+
+_Example_
+```yml
+argocd:
+  username: ${ARGOCD_USERNAME} # Now only use for instance 1
```

Comment:
As above, I think this is clear from the section above and can probably be omitted in this section.